platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationAdminMigration.java

```diff
@@ -118,4 +120,23 @@ private Path createEtcDirectory(Path exportDirectory) throws IOException {
         return etcDirectory;
     }
 
+    private String getBaseFileName(Configuration configuration) {
+        try {
+            String fileUrl = (String) configuration.getProperties()
+                    .get(FELIX_FILEINSTALL_FILENAME);
+            if (fileUrl != null) {
+                File file = new File(new URL(fileUrl).toURI());
+                return file.getName();
+            } else {
```

Comment:
It should.  This is an example of a managed service factory configuration for mime types.  If we don't find a _felix.fileinstall.filename_ we would fallback to using the _service.pid_.  Do you see any issues with that approach?
`customMimeTypes=[ \
  "xml\=text/xml;id\=csw", \
  ]
felix.fileinstall.filename="file:/Users/danielfigliola/projects/ddf-projects/ddf/distribution/ddf/target/ddf-2.11.0-SNAPSHOT/etc/DDF_Custom_Mime_Type_Resolver-csw.config"
name="CSW\ Content\ Resolver"
priority=I"10"
schema="http://www.opengis.net/cat/csw/2.0.2"
service.factoryPid="DDF_Custom_Mime_Type_Resolver"
service.pid="DDF_Custom_Mime_Type_Resolver.c8df0cfb-77ce-4bf5-ad6f-dc37c3e85ea8"`